pubsub/rabbitpubsub/rabbit.go

```diff
@@ -576,7 +576,7 @@ func (s *subscription) ReceiveBatch(ctx context.Context, maxMessages int) ([]*dr
 
 	// Get up to maxMessages waiting messages, but don't take too long.
 	var ms []*driver.Message
-	maxTime := time.Tick(50 * time.Millisecond)
+	maxTime := time.NewTimer(50 * time.Millisecond)
```

Comment:
Nice catch, thanks.

It doesn't look like we need the `for` loop here anymore; all 3 of the `select` clauses have a `return`. Can we get rid of the `for` loop, delete `maxTime` and use `time.After` in the last `select` clause, and move the `// Timed out` comment + `return` out of the `select`?

Reply:
The middle case doesn't always return (if `ok` and `len(ms) < maxMessages`)

Reply:
Ah, you are right, that is subtle.